src/py/flwr/server/strategy/dp_strategy_wrapper_fixed_clipping.py

```diff
@@ -0,0 +1,171 @@
+        noise_multiplier: float,
+        clipping_norm: float,
+        num_sampled_clients: int,
+    ) -> None:
+        super().__init__()
+
+        self.strategy = strategy
+
+        if noise_multiplier < 0:
+            raise ValueError("The noise multiplier should be a non-negative value.")
+
+        if clipping_norm <= 0:
+            raise ValueError("The clipping norm should be a positive value.")
+
+        if num_sampled_clients <= 0:
+            raise ValueError(
+                "The number of sampled clients should be a positive value."
+            )
+
+        self.noise_multiplier = noise_multiplier
+        self.clipping_norm = clipping_norm
+        self.num_sampled_clients = num_sampled_clients
+
+        self.current_round_params: NDArrays = []
+
+    def __repr__(self) -> str:
+        """Compute a string representation of the strategy."""
+        rep = "DP Strategy Wrapper with Server Side Fixed Clipping"
+        return rep
+
+    def initialize_parameters(
+        self, client_manager: ClientManager
+    ) -> Optional[Parameters]:
+        """Initialize global model parameters using given strategy."""
+        return self.strategy.initialize_parameters(client_manager)
+
+    def configure_fit(
+        self, server_round: int, parameters: Parameters, client_manager: ClientManager
+    ) -> List[Tuple[ClientProxy, FitIns]]:
+        """Configure the next round of training."""
+        self.current_round_params = parameters_to_ndarrays(parameters)
+        return self.strategy.configure_fit(server_round, parameters, client_manager)
+
+    def configure_evaluate(
+        self, server_round: int, parameters: Parameters, client_manager: ClientManager
+    ) -> List[Tuple[ClientProxy, EvaluateIns]]:
+        """Configure the next round of evaluation."""
+        return self.strategy.configure_evaluate(
+            server_round, parameters, client_manager
+        )
+
+    def aggregate_fit(
+        self,
+        server_round: int,
+        results: List[Tuple[ClientProxy, FitRes]],
+        failures: List[Union[Tuple[ClientProxy, FitRes], BaseException]],
+    ) -> Tuple[Optional[Parameters], Dict[str, Scalar]]:
+        """Compute the updates, clip them, and pass them to the child strategy for
+        aggregation.
+
+        Afterward, add noise to the aggregated parameters.
+        """
+        if failures:
+            return None, {}
+
+        # Extract all clients' model params
+        clients_params = [
+            parameters_to_ndarrays(fit_res.parameters) for _, fit_res in results
+        ]
+
+        # Compute and clip the updates
+        for client_param in clients_params:
+            compute_clip_model_update(
+                client_param, self.current_round_params, self.clipping_norm
+            )
+
+        # Update the results with the new params
+        for res, params in zip(results, clients_params):
+            res[1].parameters = ndarrays_to_parameters(params)
```

Comment:
true, but it would be. `res.parameters`. right?
        ```
for _, res in results:
            param = parameters_to_ndarrays(res.parameters)
            # compute and clip update
            compute_clip_model_update(param, self.current_round_params, self.clipping_norm)
            # convert back to parameters
            res.parameters = ndarrays_to_parameters(param)

```

Reply:
yes, like that

Reply:
Cool, I changed it.